Use empty class path for base layers with only java base modules (#399)
graalvm/graalvm-demos: native-image/microservices/spring-hello-rest-maven-layered/README.md

            </executions>
            <configuration>
                <imageName>libjavabaselayer</imageName>

>                 <classesDirectory>null</classesDirectory>

Reviewer: See https://github.com/graalvm/graalvm-demos/pull/399/changes#r3512902135

It will also create the `libjavabaselayer.so` shared library which is a run time dependency for the application layer.
Note also that you use `install` instead of `package` to ensure that the base layer JAR is installed in the `.m2` cache as it will be needed by the application build later.

> The `<classesDirectory>null</classesDirectory>` configuration is used to exclude the main jar from the image. In this case, we set it to a directory that does not exist.

Reviewer 2: ```suggestion
The `<classesDirectory>null</classesDirectory>` configuration excludes the main JAR from the image by specifying a directory that does not exist.
```

OR 
```
The `<classesDirectory>null</classesDirectory>` configuration excludes the main JAR from the image. The value `null` specifies a directory that does not exist.
```
